sql/service_wsrep.cc

```diff
@@ -187,39 +187,28 @@ extern "C" my_bool wsrep_thd_is_SR(const THD *thd)
     thd->wsrep_cs().transaction().state() == wsrep::transaction::s_executing;
 }
 
-extern "C" void wsrep_handle_SR_rollback(THD *bf_thd,
+extern "C" void wsrep_handle_SR_rollback(THD *bf_thd __attribute__((unused)),
                                          THD *victim_thd)
 {
+  /*
+    We should always be in victim_thd context, either client session is
+    rolling back or rollbacker thread should be in control.
+  */
   DBUG_ASSERT(victim_thd);
-  DBUG_ASSERT(wsrep_thd_is_SR(victim_thd));
```

Comment:
I think we could keep DBUG_ASSERT(wsrep_thd_is_SR(victim_thd)).
Is there a reason to remove it?

Reply:
When this function is called there is always a condition like  is_wsrep() && wsrep_thd_is_SR(mysql_thd) before but I restore it to avoid unsupported usage case when new place is added.